### PR TITLE
Maurice/mailfunction

### PR DESCRIPTION
**Pull request by Barnes2197**
Added my simple mail function only sends from my email address to mines or jehfs and a sample test. Just for testing amazon ses and pug templates

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import createPubsub from './createPubSub';
 import app from './app';
+import 'env';
```
Comment (Jehfkemsy) [inline src/server.ts]: `env` should be the first thing you import 
ref: https://github.com/motdotla/dotenv/issues/133#issuecomment-255298822

File: src/env.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { config } from 'dotenv';
+
+config({ path: '.env' });
```
Comment (Jehfkemsy) [inline src/env.ts]: Here you might wanna validate to make sure all `env` variables are available... you can use [join](https://hapi.dev/family/joi/)
You should create a `/models` folder for the `env` schema and and import it into `env.ts`

File: src/.env.example [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Amazon SES credentials
+ACCESSKEYID:
+SECRETACCESSKEY:
```
Comment (Jehfkemsy) [inline src/.env.example]: `.env` uses the `=` sign, not `:`

File: src/Services/MailService.ts [hunk 4]
```diff
@@ -1,5 +1,66 @@
+import nodemailer, { createTestAccount } from 'nodemailer';
+import AWS from 'aws-sdk';
+import { pugEngine } from 'nodemailer-pug-engine';
+import Mail from 'nodemailer/lib/mailer';
+import '../env';
```
Comment (Jehfkemsy) [inline src/Services/MailService.ts]: you only need to import `env` once in the root file (aka your app entry point)

File: src/Services/MailService.ts [hunk 5]
```diff
@@ -1,5 +1,66 @@
+import nodemailer, { createTestAccount } from 'nodemailer';
+import AWS from 'aws-sdk';
+import { pugEngine } from 'nodemailer-pug-engine';
+import Mail from 'nodemailer/lib/mailer';
```
Comment (Jehfkemsy) [inline src/Services/MailService.ts]: is there a reason why you have two separate imports for `nodemailer`?

**Comment by Jehfkemsy** [comment]
You also need to run `npm run format` to fix the CI errors

**Comment by Barnes2197** [comment]
> You also need to run `npm run format` to fix the CI errors

I get an error everytime I run this comman @Jehfkemsy

File: src/.env.example (hunk 3 above)
Comment (Barnes2197) [reply src/.env.example]: Yeah I'm going to change it. Forgot to change in example

File: src/Services/MailService.ts (hunk 4 above)
Comment (Barnes2197) [reply src/Services/MailService.ts]: Going to see if I can combine the 2 imports. Honestly don't know why they were seperated.

File: src/Services/MailService.ts (hunk 5 above)
Comment (Barnes2197) [reply src/Services/MailService.ts]: Going to see if I can combine the 2 imports. Honestly don't know why they were seperated.

File: src/env.ts (hunk 2 above)
Comment (Barnes2197) [reply src/env.ts]: Wouldn't I only need one mode for the .env????

**Pull request closed without merging**